src/App.tsx

```diff
@@ -7,26 +7,61 @@ import CallbackHandler from "./components/authentication/callback_handler";
 import Nav from "./components/navigation/nav";
 import EngagementForm from "./routes/engagement_form";
 
-import { SessionProvider } from './context/session_context';
+import SessionContext, { SessionProvider } from "./context/session_context";
+import ConfigContext, { ConfigProvider } from "./context/config_context";
+import Axios from "axios";
+import { EngagementFormProvider } from "./context/engagement_form_context";
 
 const App = () => {
   return (
     <Router>
-      <SessionProvider>
-        <Page
-          header={<PageHeader />}
-          sidebar={<PageSidebar isNavOpen theme="dark" nav={<Nav />} />}
-          style={{ height: "100vh" }}
-        >
-          <Switch>
-            <Route exact path="/" component={EngagementForm} />
-            <PrivateRoute path="/private" component={EngagementForm} />
-            <Route path="/auth_callback" component={CallbackHandler} />
-          </Switch>
-        </Page>
-      </SessionProvider>
+      <ConfigProvider>
+        <SessionProvider>
+          <Routes />
+        </SessionProvider>
+      </ConfigProvider>
     </Router>
   );
 };
 
+const Routes = () => {
+  const configContext = useContext(ConfigContext);
+  const sessionContext = useContext(SessionContext);
+  useEffect(() => {
+    Axios.get(`${process.env.PUBLIC_URL}/config.json`).then(({ data }) => {
```

Comment:
Wrapping my head around everything. Is the thought that this is empty and will resolve to simply `/config.json` when built in a container? But we can use it to set a file url or something for running locally?

Reply:
Yep, so all you have to do is drop a `config.json` into the public folder when running on dev, and the file will be available when running on dev. During the build process, CRA copies everything from the `public` folder directly to the `build` folder. So, all we should need to do in different environments is mount a volume to `${build_output_dir}/config.json`.